update tokio support

sp3d opened the pull request:
This is just a forward-port of the existing tokio branch to the current state of the tokio libraries. I'm successfully depending on this in an application, so after merging into the tokio branch I'd be interested in what additional is desired before the tokio branch is considered acceptable to merge into master. Thanks!

Commit:
update tokio support

bkchr commented on src/tokio.rs:
```diff
@@ -69,54 +69,56 @@ impl io::Write for NetlinkSocket {
             }
-            if let Some(pkt) = NetlinkPacket::new(slice) {
-                //println!("{:?} slice: {}", pkt, slice.len());
+            if let Some(pkt) = NetlinkPacket::new(buf) {
```
Maybe use an expect here.

sp3d replied:
I think it's better not to panic, since it's less trouble for consumers. I've amended with an Err return for this case, and also removed the println calls.

bkchr reviewed (changes requested):
Just change this if to use a `expect` and then I'm merging this pull request.

The next step would be to remove these println and  rebase to master. After these two things are done, you can open a pr for merging into master :)

Commit:
clean up tokio support

implement error handling and remove Decoder::decode_eof implementation, since the default one suffices for the netlink protocol
